radixdlt-core/radixdlt/src/main/java/com/radixdlt/client/service/HighLevelApiService.java

```diff
@@ -64,8 +64,15 @@ public int getUniverseMagic() {
 	}
 
 	public Result<TokenDefinitionRecord> getNativeTokenDescription() {
-		return clientApiStore.getTokenSupply(nativeTokenDefinition.getRri())
-			.map(supply -> TokenDefinitionRecord.from(nativeTokenDefinition, supply));
+		return Result.fromOptional(
+			immutableIndex.loadRriId(null, RriId.nativeToken()),
+			"Unable to find native token"
+		)
+			.map(p -> (TokenDefinitionParticle) p)
```

Comment:
Perhaps `TokenDefinitionParticle.class::cast` would be little bit more readable.